### PR TITLE
Renaming all occurrences of test-procedure to procedure

### PR DESCRIPTION
**Pull request by beaioun**
### Description
Renaming all occurrences of the name test-procedure and test_procedure to procedure, class name TestProcedure to Procedure

### Issues Resolved
issue #308

### Testing
- [ ] New functionality includes testing

### Describe how this change was tested
a unit test was run and more tests will be conducted for the following commit on other test-procedure names.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/metrics.py [hunk 1]
```diff
@@ -453,7 +453,7 @@ def open(self, test_ex_id=None, test_ex_timestamp=None, workload_name=None,\
         :param test_ex_id: The test execution id. This attribute is sufficient to uniquely identify a test_execution.
         :param test_ex_timestamp: The test execution timestamp as a datetime.
         :param workload_name: Workload name.
-        :param test_procedure_name: TestProcedure name.
+        :param test_procedure_name: Procedure name.
```
Comment (IanHoang) [inline osbenchmark/metrics.py:456]: test_procedure_name -> procedure_name

File: tests/workload/loader_test.py [hunk 2]
```diff
@@ -1307,10 +1307,10 @@ def test_sets_absolute_path(self, path_exists):
         cfg = config.Config()
         cfg.add(config.Scope.application, "benchmarks", "local.dataset.cache", "/data")
 
-        default_test_procedure = workload.TestProcedure("default", default=True, schedule=[
+        default_test_procedure = workload.Procedure("default", default=True, schedule=[
             workload.Task(name="index", operation=workload.Operation("index", operation_type=workload.OperationType.Bulk), clients=4)
         ])
-        another_test_procedure = workload.TestProcedure("other", default=False)
+        another_test_procedure = workload.Procedure("other", default=False)
         t = workload.Workload(name="u", test_procedures=[another_test_procedure, default_test_procedure],
```
Comment (IanHoang) [inline tests/workload/loader_test.py:1314]: You'll need to also change occurrences like `test_procedures`, `another_test_prcedure`, `default_test_procedure`. However, I think your approach is good if you plan to do a commit for each variation in this PR. For example, in the latest commit, you're renaming only `TestProcedure` -> `Procedure`. In subsequent commits, you can dedicate a commit to renaming instances of `test_procedure` or `test-procedure`.

File: tests/workload/loader_test.py [hunk 3]
```diff
@@ -1942,7 +1942,7 @@ def test_parse_duplicate_implicit_task_names(self):
         reader = loader.WorkloadSpecificationReader()
         with self.assertRaises(loader.WorkloadSyntaxError) as ctx:
             reader("unittest", workload_specification, "/mappings")
-        self.assertEqual("Workload 'unittest' is invalid. TestProcedure 'default-test_procedure' contains multiple tasks"
+        self.assertEqual("Workload 'unittest' is invalid. Procedure 'default-test_procedure' contains multiple tasks"
```
Comment (IanHoang) [inline tests/workload/loader_test.py:1945]: `default-test_procedure` will also be corrected when you do a replace for `test_procedure` -> `procedure`

**Review by IanHoang — CHANGES_REQUESTED** [review]
Thanks for this! This is the final piece to the puzzle for our renaming components initiative. I like how you're creating different commits to address the different variations of Test Procedures.

**Comment by beaioun** [comment]
> Thanks for this! This is the final piece to the puzzle for our renaming components initiative. I left a few comments. I like how you're creating different commits to address the different variations of Test Procedures.

Yes Ian, that's the plan. I will finish working on the rest of naming tomorrow. And maybe test about its interaction with the workload to see if it will run into any issues.

**Comment by beaioun** [comment]
Hey, @IanHoang I did manage to rename every occurrence in the repo but failed the integration test over this Error: `[ERROR] Cannot execute-test. Error in test execution orchestrator (Workload 'geonames' is invalid. You must define 'procedure', 'procedures', or 'schedule' but none is specified.)` or something similar. I couldn't trace the root cause of this error but it seems to lead to workload/loader.py which is expected as we have discussed.
The next step is maybe to make changes to the workload repo in a safe manner.

**Comment by IanHoang** [comment]
Update: Currently working with Mingyang (@beaioun) on testing these changes out

**Comment by IanHoang** [comment]
Closing this PR as a duplicate has been created

**Pull request closed without merging**